CodingGuidelines.md

```diff
@@ -61,12 +191,76 @@ private Foo()
 }
 ```
 
+---
+
+Public classes, structs, and enums should all go in their own files.
+---
+
+>If the class, struct, or enum can be made private then it's okay to be included in the same file.  This avoid compilations issues with Unity and ensure that proper code abstraction occurs, it also reduces conflicts and breaking changes when code needs to change.
+
+### Don't:
+
+```
+public class MyClass
+{
+    public struct MyStruct() { }
+    public enum MyEnumType() { }
+    public class MyNestedClass() { }
+}
+```
+
+### Do:
+
+ ```
+ // Private references for use inside the class only
+public class MyClass
+{
+    private struct MyStruct() { }
+    private enum MyEnumType() { }
+    private class MyNestedClass() { }
+}
+ ```
+
+ ### Do:
+
+ ```
+ // Public Struct / Enum definitions for use in your class.  Try to make them generic for reuse.
+MyStruct.cs
+public struct MyStruct
+{
+    public string Var1;
+    public string Var2;
+}
+
+MyEnumType.cs
+public enum MuEnumType
+{
+    Value1,
+    Value2 // <- note, no "," on last value to denote end of list.
+}
+
+MyClass.cs
+public class MyClass
+{
+    private MyStruct myStructreference;
+    private MyEnumType myEnumReference;
+}
+ ```
+---
+
+## Best Practices, including Unity recommendations
+Some of the target platforms of this project require us to take performance into consideration.  What this in mind we should always be careful of allocating memory in frequently called code in tight update loops or algorithms.
```

Comment:
What -> With

Reply:
@StephenHodgson, I think the What -> With fix is close to the last thing we need changed.